x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/common/ProgressListenableActionFutureTests.java

```diff
@@ -233,9 +233,46 @@ public void testListenerCalledImmediatelyWhenProgressReached() {
         assertThat(future.isDone(), is(true));
     }
 
+    public void testLongConsumerCalledOnProgressUpdate() {
+        // min length of 2 to have at least one progress update before reaching the end
+        long length = randomLongBetween(2L, ByteSizeUnit.TB.toBytes(1L));
+        long start = randomLongBetween(Long.MIN_VALUE, Long.MAX_VALUE - length);
+        long end = start + length;
+
+        var consumed = new HashSet<Long>();
+        var future = new ProgressListenableActionFuture(
+            start,
+            end,
+            p -> assertThat("LongConsumer should not consumed the same  value twice", consumed.add(p), equalTo(true))
+        );
+
+        long position = start;
+        for (int i = 0; i < 25 && position < end - 1L; i++) {
```

Comment:
Nit: Maybe a comment about why `25` is used could be helpful.

Reply:
This is just a number of iterations, I used a local variable in 16052d4fa5cf003d0243518f41c08c51a8002314